docs/proposals/Annotation-Based-Artifact-Filtration.md

```diff
@@ -0,0 +1,131 @@
+
+This kind of filtering strategy is best used on artifacts that are rapidly changing, for example Vulnerability Assessment artifacts that are immedietly oudated once a new artifact is pushed to the registry.
+
+## Annotation Value based filteration
+
+Ratify could perform string comparison checks in order to evaluate if a given arifact should be evaluted by a given verifier or not, the flitering would be made by checking if a given annotation
+is part of a list of approved annotation values.
+
+To illustrate an example usage of this kind of filtration we could take a look at the following notary artifact manifest:
+
+```json
+{
+  "schemaVersion": 2,
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.config.v1+json",
+      "size": 7023,
+      "digest": "sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b864",
+      "signatures": [
+        {
+          "mediaType": "applicaiton/vnd.cncf.notary.x509-signature.config.v2",
+          "digest": "sha256:134c7fe821b9d359490cd009ce7ca322453f4f2d0",
+          "size": 1337,
+          "annotations": {
+            "org.opencontainers.image.signature.fingerprint": "43:51:43:a1:b5:fc:8b:b7:0a:3a:a9:b1:0f:66:73:a8"
+          }
+        }
+      ]
+    }
+  ]
+}
+```
+
+We could define a notation verifier configuration that would require the artifact manifest to contain the annotation `org.opencontainers.image.signature.fingerprint` and that is should have a known fingerprint value. this would enable a scenario where the user configures three notation verifiers, one in each namespace:
+* Namepsace A -> Fingerprint should match [X]
+* Namespace B -> Fingerprint should match [Y]
+* Namespace C -> Fingerprint should match either [X, Y]
+
+## User experiences
+
+This section describes the experience that users interact with Ratify using the proposed solution. In summary, the propsed solution suggest we should allow for filtering of artifact based on annotations, and as such the following section describes how the customer would configure the filtration.
+
+Seeing that filteration is unique to each verifier, it should be configured in the verifier itself, as such, in order to maintain backwards compatability, it is important to note that if no filteration is configured the default behavior would be to evaluate all artifacts.
+
+### Defining artifact age based filtering
+
+To support artifact age based filtering, we would add an additional field to the verifier configuration:
+
+```yaml
+apiVersion: config.ratify.deislabs.io/v1beta1
+kind: Verifier # NamespacedVerifier has the same spec.
+metadata:
+  name: test-verifier
+spec:
+  name: # REQUIRED: [string], the unique type of the verifier (notation, cosign)
+  artifactType: # REQUIRED: [string], comma seperated list, artifact type this verifier handles
+  artifactAgeFilter: # Optional: [string], one of the following 'latest' to verify only the last artifact, or all to verify all artifacts. defaults to all. 
```

Comment:
will the value of `artifactAgeFilter` be a define set of string? an expression or key:value pair

Reply:
It would be as a single string, either "latest" or "all"